bot/exts/evergreen/bookmark.py

```diff
@@ -88,11 +89,16 @@ async def send_reaction_embed(
     async def bookmark(
         self,
         ctx: commands.Context,
-        target_message: WrappedMessageConverter,
+        target_message: t.Optional[WrappedMessageConverter],
         *,
         title: str = "Bookmark"
     ) -> None:
         """Send the author a link to `target_message` via DMs."""
+        if not target_message:
+            if not ctx.message.reference:
+                raise commands.UserInputError("You must either provide a valid message to bookmark, or reply to one.")
+            target_message = await ctx.fetch_message(ctx.message.reference.message_id)
```

Comment:
Could we try to get this from the cache before making an API request?

Reply:
Ideally if lance is using discord.py 1.6.0 or above we can just use the resolved reference so we never need to fetch from the API

Reply:
currently it's 1.5.1